### PR TITLE
Rails 6 spree 4

### PR DESCRIPTION
**Pull request by PSzyszkaSpark**
Fixes https://spark-solutions.atlassian.net/wiki/spaces/SOS/pages/848986120/Updating+Extensions+to+Rails+6+and+Spree+4

### CODE REVIEW

File: app/models/spree/product_decorator.rb [hunk 1]
```diff
@@ -1,106 +1,117 @@
...
 
-  def format_name(name)
-    name.to_s.downcase.gsub(' ', '_').pluralize
+    def format_name(name)
+      name.to_s.downcase.tr(' ', '_').pluralize
+    end
   end
 end
+
+Spree::Product.prepend(Spree::ProductDecorator)
```
Comment (pmilewski) [inline app/models/spree/product_decorator.rb]: `::Spree::Product.prepend(Spree::ProductDecorator)`

File: app/models/spree/product_decorator.rb [hunk 2]
```diff
@@ -1,106 +1,117 @@
...
+  module ProductDecorator
+    def self.prepended(base)
+      base.has_many :relations, -> { order(:position) }, as: :relatable
+
+      # When a Spree::Product is destroyed, we also want to destroy all
+      # Spree::Relations "from" it as well as "to" it.
+      base.after_destroy :destroy_product_relations
+
+      # Returns all the Spree::RelationType's which apply_to this class.
+      def base.relation_types
```
Comment (btolarz) [inline app/models/spree/product_decorator.rb]:
```module Foo
  def self. prepended base
    base.extend ClassMethods
  end

  module ClassMethods
    def bar
      'bar'
    end
  end
end
```

File: app/models/spree/product_decorator.rb [hunk 3]
```diff
@@ -1,106 +1,117 @@
...
-    else
-      relations_for_relation_type(relation_type)
+    # Decides if there is a relevant Spree::RelationType related to this class
+    # which should be returned for this method.
+    #
+    # If so, it calls relations_for_relation_type. Otherwise it passes
+    # it up the inheritance chain.
+    def method_missing(method, *args)
+      # Fix for Ruby 1.9
+      raise NoMethodError if method == :to_ary
```
Comment (btolarz) [inline app/models/spree/product_decorator.rb]: In spree we have more things witch will not work with ruby 1.9, you can remove this line.

File: lib/spree_related_products.rb [hunk 4]
```diff
@@ -4,3 +4,4 @@
 require 'spree_related_products/version'
 require 'sass/rails'
 require 'spree_extension'
+require 'deface'
```
Comment (btolarz) [inline lib/spree_related_products.rb]: We should drop deface support I think.

Comment (PSzyszkaSpark) [reply lib/spree_related_products.rb]: If I drop deface support than we can't override views from spree with it and I would need to do it manually by duplicating all the code from the view.

**Pull request merged**